### PR TITLE
Shared lib directory for jpscore and jpsreport

### PR DESCRIPTION
**Pull request by JetteSchumann**
In order to use the logger in jpscore and jpsreport without accessing the sources from another project a common parent folder `libshared` is created. Logger files have been moved.

Including the lib does not work so far.
As I am not that familiar with cmake, help is needed.

### CODE REVIEW

**Comment by JetteSchumann** [comment]
Thanks @schroedtert 

Think we need to wait for @zeroset to have a look at it.

**Comment by Ozaq** [comment]
@JetteSchumann I can have a look at the CMake changes, but since this is still a draft I will wait until you tell me its fine to look at.

**Comment by Ozaq** [comment]
@JetteSchumann sorry for the delay didn't make it yesterday :/

File: libshared/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+set(header_files
+   Logger.h
+)
+
+add_library(libshared STATIC
+        ${source_files}
+        ${header_files}
+)
+
```
[the remark this answers is not in the document]
Comment (JetteSchumann) [reply libshared/CMakeLists.txt:17]: Thanks. This is much better than the relative path ;)

File: libshared/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+target_link_libraries(libshared PUBLIC
+        spdlog::spdlog
+        fmt::fmt
+        git-info
```
[the remark this answers is not in the document]
Comment (JetteSchumann) [reply libshared/CMakeLists.txt:21]: Done. Was not sure what git-info is needed for...

File: libshared/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+)
+
+target_compile_options(libshared PRIVATE
+        ${COMMON_COMPILE_OPTIONS}
+)
+
+target_compile_definitions(libshared PUBLIC
+        JPSCORE_VERSION="${PROJECT_VERSION}"
+)
```
[the remark this answers is not in the document]
Comment (JetteSchumann) [reply libshared/CMakeLists.txt:30]: Done

File: libshared/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+set(source_files
+   Logger.cpp
+)
+
+set(header_files
+   Logger.h
+)
+
+add_library(libshared STATIC
```
[the remark this answers is not in the document]
Comment (JetteSchumann) [reply libshared/CMakeLists.txt:13]: Thanks. Done.

File: libcore/CMakeLists.txt [hunk 5]
```diff
@@ -213,9 +211,11 @@ target_link_libraries(core
     spdlog::spdlog
     fmt::fmt
     git-info
+    libshared
 )
 target_include_directories(core PUBLIC
     ${CMAKE_CURRENT_SOURCE_DIR}/src
+        ${CMAKE_CURRENT_SOURCE_DIR}/../libshared
```
[the remark this answers is not in the document]
Comment (JetteSchumann) [reply libcore/CMakeLists.txt:218]: Done

File: libshared/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        spdlog::spdlog
+        fmt::fmt
+        git-info
+)
+
+target_compile_options(libshared PRIVATE
+        ${COMMON_COMPILE_OPTIONS}
+)
+
+target_compile_definitions(libshared PUBLIC
```
[the remark this answers is not in the document]
Comment (JetteSchumann) [reply libshared/CMakeLists.txt:28]: Done

**Comment by JetteSchumann** [comment]
Thanks @Ozaq.

PR solves #793

**Pull request merged**